core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
Two comments:

- This handling should be moved to SnappyConf as mentioned in the other PR.

- Secondly the Spark property is a inheratable thread-local but we need the property to be per SnappySession. In particular this will not work for routed queries because the setter thread may be different that the threads used in subsequent queries. So this means that we cannot be setting the SparkContext property here and be done. Instead what we should do is:
  - keep the check for pool existence but then set a variable in SnappyConf and public getter to read it, like: sessionPool which defaults to the "default" pool (or None can indicate the same)
  - in key set action set to the new given pool and in unset case revert to "default" (or None)
  - read the current pool from SnappyConf in SnappySession.executeSQL, evaluatePlan and CachedDataFrame.collectWithHandler and set/clear the pool local property there on SparkContext. Alternatively it will be better to set it in just CodegenSparkFallback.executeWithFallback in before "f(plan)" calls assuming that threads won't be switched between plan execution and collect. This will allow it to work for sql as well as API.